controllers/reconcilers/kserve_istio_smm_reconciler.go

```diff
@@ -0,0 +1,147 @@
+package reconcilers
+
+import (
+	"context"
+	"errors"
+
+	"github.com/go-logr/logr"
+	kservev1beta1 "github.com/kserve/kserve/pkg/apis/serving/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "maistra.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/opendatahub-io/odh-model-controller/controllers/comparators"
+	"github.com/opendatahub-io/odh-model-controller/controllers/constants"
+	"github.com/opendatahub-io/odh-model-controller/controllers/processors"
+	"github.com/opendatahub-io/odh-model-controller/controllers/resources"
+	"github.com/opendatahub-io/odh-model-controller/controllers/utils"
+)
+
+type KserveServiceMeshMemberReconciler struct {
+	SingleResourcePerNamespace
+	client         client.Client
+	smmHandler     resources.ServiceMeshMemberHandler
+	deltaProcessor processors.DeltaProcessor
+}
+
+func NewKserveServiceMeshMemberReconciler(client client.Client) *KserveServiceMeshMemberReconciler {
+	return &KserveServiceMeshMemberReconciler{
+		client:         client,
+		smmHandler:     resources.NewServiceMeshMember(client),
+		deltaProcessor: processors.NewDeltaProcessor(),
+	}
+}
+
+func (r *KserveServiceMeshMemberReconciler) Reconcile(ctx context.Context, log logr.Logger, isvc *kservev1beta1.InferenceService) error {
+	log.V(1).Info("Verifying that the namespace is enrolled to the mesh")
+
+	// Create Desired resource
+	desiredResource, err := r.createDesiredResource(ctx, isvc)
+	if err != nil {
+		return err
+	}
+
+	// Get Existing resource
+	existingResource, err := r.getExistingResource(ctx, log, isvc.Namespace)
+	if err != nil {
+		return err
+	}
+
+	// Process Delta
+	if err = r.processDelta(ctx, log, desiredResource, existingResource); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *KserveServiceMeshMemberReconciler) Cleanup(ctx context.Context, log logr.Logger, namespace string) error {
+	existingSMM, getError := r.getExistingResource(ctx, log, namespace)
+	if getError != nil {
+		return getError
+	}
+
+	return r.processDelta(ctx, log, nil, existingSMM)
+}
+
+func (r *KserveServiceMeshMemberReconciler) createDesiredResource(ctx context.Context, isvc *kservev1beta1.InferenceService) (*v1.ServiceMeshMember, error) {
+	smcpName, smcpNamespace := utils.GetIstioControlPlaneName(ctx, r.client)
+
+	return &v1.ServiceMeshMember{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      constants.ServiceMeshMemberName,
+			Namespace: isvc.Namespace,
+			Labels: map[string]string{
+				"app.kubernetes.io/name":       "odh-model-controller",
+				"app.kubernetes.io/component":  "kserve",
+				"app.kubernetes.io/part-of":    "odh-model-serving",
+				"app.kubernetes.io/managed-by": "odh-model-controller",
```

Comment:
Are there reusable constants for these?

Reply:
Not at the moment: https://github.com/opendatahub-io/odh-model-controller/blob/main/controllers/constants/constants.go